packages/clerk-js/src/ui/components/UserProfile/SecurityPage.tsx

```diff
@@ -14,13 +14,21 @@ import { PasswordSection } from './PasswordSection';
 import { getSecondFactors } from './utils';
 
 export const SecurityPage = withCardStateProvider(() => {
-  const { attributes, instanceIsPasswordBased } = useEnvironment().userSettings;
+  const { attributes, instanceIsPasswordBased, enterpriseSSO } = useEnvironment().userSettings;
   const card = useCardState();
   const { user } = useUser();
   const showPassword = instanceIsPasswordBased;
   const showPasskey = attributes.passkey?.enabled;
   const showMfa = getSecondFactors(attributes).length > 0;
   const showDelete = user?.deleteSelfEnabled;
+  const showEnterpriseAccounts = user && enterpriseSSO.enabled;
+
+  const shouldAllowIdentificationCreation =
```

Comment:
We should pass this logic of `shouldAllowIdentificationCreation` to the user profile context (`useUserProfileContext`) in order to not keep repeating in every single inner page/component.